lib/registry/registry.rb

```diff
@@ -49,16 +51,15 @@ def paginate_doget(url)
         response = doget(url)
         yield response
 
-        link_header = response.headers[:link]
-        break unless link_header
-
-        url = parse_link_header(link_header)[:next]
+        link_header = response.headers[:link] or break
+        next_url = parse_link_header(link_header)[:next] or break
+        url = URI.join(response.request.url, next_url) # Interpret the next link relative to the current URL.
       end
     end
 
     def search(query = '')
       all_repos = []
-      paginate_doget('/v2/_catalog') do |response|
+      paginate_doget('v2/_catalog') do |response|
```

Comment:
Why remove the leading `/` here (and in all other cases)? Is this because it used to just `@base_uri + path` so it always added it? And thus if the base was `http://foo.com/a/b` then this would give `http://foo.com/a/b/v2/_catalog`.  But with `URI.join`, it will treat the leading `/` as an absolute path, and so unless you remove the leading `/`, you would end up with `http://foo.com/v2/_catalog` instead of `http://foo.com/a/b/v2/_catalog`?

Reply:
Exactly. In convential URL semantics, paths starting with a `/` are absolute.